File: Calculator.java
import java.util.Scanner;

public class Calculator {
    public static void main(String[] args) {
        Scanner ip = new Scanner(System.in);
        System.out.print("Enter two numbers to perform an arithmetic operation of your choice\n ");
        // Input two numbers
        System.out.print("Enter the first number: ");
        double num1 = ip.nextDouble();

        System.out.print("Enter the second number: ");
        double num2 = ip.nextDouble();

        // Choose arithmetic operation
        System.out.println("Choose an operation from the list given below");
        System.out.println("1. Addition");
        System.out.println("2. Subtraction");
        System.out.println("3. Multiplication");
        System.out.println("4. Division");

        System.out.print("Enter your choice (1-4): ");
        int choice = ip.nextInt();  
        double result = 0.0;

        if (choice == 1) {
            result = num1 + num2;
        } else if (choice == 2) {
            result = num1 - num2;
        } else if (choice == 3) {
            result = num1 * num2;
        } else if (choice == 4) {
            if (num2 != 0) {
                result = num1 / num2;
            } else {
                System.out.println("Error: Cannot divide by zero.");
                return; // Exit the program
            }
        } else {
            System.out.println("Invalid choice. Please choose a number between 1 and 4.");
            return; // Exit the program
        }

        // Display the result
        System.out.println("Result: " + result);

        // Close the scanner
        ip.close();
    }
}

File: calculator.cpp
#include <iostream>

using namespace std;

int main() {
    // Declare variables to store user input and result
    double num1, num2, result;
    int choice;

    // Prompt the user to enter two numbers
    cout << "Enter first number: ";
    cin >> num1;

    cout << "Enter second number: ";
    cin >> num2;

    // Display the menu
    cout << "Choose an operation:" << endl;
    cout << "1. Addition" << endl;
    cout << "2. Subtraction" << endl;
    cout << "3. Multiplication" << endl;
    cout << "4. Division" << endl;
    cout << "Enter your choice (1-4): ";
    cin >> choice;

    // Perform arithmetic operations based on user choice
    switch (choice) {
        case 1:
            result = num1 + num2;
            cout << "Result: " << num1 << " + " << num2 << " = " << result << endl;
            break;
        case 2:
            result = num1 - num2;
            cout << "Result: " << num1 << " - " << num2 << " = " << result << endl;
            break;
        case 3:
            result = num1 * num2;
            cout << "Result: " << num1 << " * " << num2 << " = " << result << endl;
            break;
        case 4:
            // Check for division by zero
            if (num2 != 0) {
                result = num1 / num2;
                cout << "Result: " << num1 << " / " << num2 << " = " << result << endl;
            } else {
                cout << "Error: Division by zero is not allowed." << endl;
                return 1;  // Return 1 to indicate an error
            }
            break;
        default:
            cout << "Error: Invalid choice. Please enter a number between 1 and 4." << endl;
            return 1;  // Return 1 to indicate an error
    }

    return 0;
}
